crates/ra_cargo_watch/src/lib.rs

```diff
@@ -259,7 +259,7 @@ pub fn run_cargo(
     let mut child = command
         .args(args)
         .stdout(Stdio::piped())
-        .stderr(Stdio::null())
+        .stderr(Stdio::piped())
```

Comment:
This I believe is technically wrong, as process pipes are of bounded capacity, and if Cargo to write a lot of output to stderr, it'll block. Hm, now that I think about it, we might actually hit this case, if some proc macro writes something to stderr (because, for example, RUST_LOG env variable is set)....

Ideally, we should use `select` here to drain both pipes, but that seems like a lot of work. What if we just check the exit status, and, if it is `1`, we log the command line? I think that should be enough to diagnose the issues.

Reply:
The 1 would not be enough. In the example screenshot, I provided the status is 101...

Reply:
The problem is that we lack the general error handling infrastructure for the cargo watcher...

Reply:
Hm, indeed. I guess the right heuristic here is "the set of messages is empty, but the return code is non-zero".

Reply:
Hm, right, I think this is a good criterion because the only successful outcome for us will be the metadata output. However, it's bad that we cannot log the actual `stderr` contents right away. Might I ask you to give a pointer to which particular `select` you referred just for my curiosity?

Reply:
compiletest uses this: https://github.com/rust-lang/rust/blob/f4c675c476c18b1a11041193f2f59d695b126bc8/src/tools/compiletest/src/read2.rs. Maybe copy it somewhere yet another time. Or just make a crate to be shared between all users of a copy of this code.

Reply:
Thanks bjorn, that is a good source of knowledge. I'd like to avoid copying that to `rust-analyzer` repo. Maybe to `ra_vfs`?

Reply:
> Or just make a crate to be shared between all users of a copy of this code.

I think this would be the best.

Reply:
Yeah, it would be great, though I don't feel strong enough on this topic... Unless anyone else does it :D !
For this use case, we can just use a simple solution of not reading the stderr here for now.